test/it/crdvalidation/etcd/specsharedconfig_test.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcd
+
+import (
+	"testing"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/test/utils"
+)
+
+// tests for etcd.spec.sharedConfig fields:
+
+// validates whether the value passed to the etcd.spec.sharedConfig.autoCompactionMode is either set as "periodic" or "revision"
+func TestValidateSpecSharedConfigAutoCompactionMode(t *testing.T) {
+	tests := []struct {
+		name               string
+		etcdName           string
+		autoCompactionMode string
+		expectErr          bool
+	}{
+		{
+			name:               "Valid autoCompactionMode #1: periodic",
+			etcdName:           "etcd-valid-1",
+			autoCompactionMode: "periodic",
+			expectErr:          false,
+		},
+		{
+			name:               "Valid autoCompactionMode #1: revision",
+			etcdName:           "etcd-valid-2",
+			autoCompactionMode: "revision",
+			expectErr:          false,
+		},
+		{
+			name:               "Invalid autoCompactionMode #1: invalid value",
+			etcdName:           "etcd-invalid-1",
+			autoCompactionMode: "frequent",
```

Comment:
```suggestion
			autoCompactionMode: "dummy",
```

`frequent` actually sounds like a possibly real compaction mode that could be supported any time in the future, so in testing we usually use explicitly garbage/invalid values to indicate that the value is invalid.